proposals/pubsub.md

```diff
@@ -0,0 +1,182 @@
+- Features like IPNS over pubsub depends on this.
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+- Measure what is the overhead on nodes that do not directly use pubsub
+- If overhead is neglegable ship a release of go-ipfs with pubsub enabled (see [ipfs/go-ipfs#6621][])
+- Enable / upgrade IPFS nodes operated by PL.
+- Raise awareness (blog, tweet, support community transition)
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+- Shipped release of go-ipfs with pubsub enabled
+- Increased number of pubsub enabled IPFS nodes in the network.
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+- Various projects (like OrbitDB, [Forest](https://github.com/forestpm/forest)) no longer have to explain with instructions how to enable pubsub.
+- Teams like fission deliver pubsub to their users requesting it.
+- Pubsub becomes more reliable due to increased number of enabled nodes. 
+
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+
+#### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+- Everyone enables pubsub already anyway.
+- Pusub prooves (more) unreliable even with increased number of enabled nodes.
+
+#### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need ?_
+
+- Alternative messaging primitives (e.g. Federated SSB like pubs)
+- [SMTP][] support in IPFS nodes (you can email any node by email address)
+
+#### Dependencies/prerequisites
+<!--List any other projects that are dependencies/prerequisites for this project that is being pitched.-->
+
+- none
+
+#### Future opportunities
+<!--What future projects/opportunities could this project enable?-->
+
+- Improving IPNS (through IPNS over pubsub)
+- DNS Service Discovery (DNS-SD) but over the global IPFS network through PubSub.
+
+## Required resources
+
+#### Effort estimate
+<!--T-shirt size rating of the size of the project. If the project might require external collaborators/teams, please note in the roles/skills section below). 
+For a team of 3-5 people with the appropriate skills:
+- Small, 1-2 weeks
+- Medium, 3-5 weeks
+- Large, 6-10 weeks
+- XLarge, >10 weeks
+Describe any choices and uncertainty in this scope estimate. (E.g. Uncertainty in the scope until design work is complete, low uncertainty in execution thereafter.)
+-->
+
+- Small to Medium
+
+> I am not really qualified to answer this question. If we find overhead reasonable, enabling it should be metter of switching defaults.
```

Comment:
It might be more clear to list out how many dev effort weeks would take for each step listed in the plan of attack including our guess for adding any necessary metrics, analyzing results, etc.

Reply:
@aschmahmann @Stebalien are probably in better position to do educated guesses here.

Reply:
We can't. Unfortunately, this is Small to Large because it depends on whether the solution is "flip a switch" or "fix pubsub".